File: Leetcode/Day-2/Java/Set Matrix Zeros.java
class Solution {
    public void setZeroes(int[][] matrix) {

        HashSet<Integer> cols = new HashSet<>();
        HashSet<Integer> rows = new HashSet<>();

        for(int i = 0; i < matrix.length; i++){
            for (int j = 0; j< matrix[0].length; j++){
                if (matrix[i][j] == 0){
                    cols.add(j);
                    rows.add(i);
                }
            }
        }

        for(int i = 0; i < matrix.length; i++){
            for (int j = 0; j < matrix[0].length; j++){
                if (cols.contains(j) || rows.contains(i)){
                    matrix[i][j] = 0;
                }
            }
        }
    }
}

File: Leetcode/Day-2/Python/Set Matrix Zeros.py
class Solution(object):
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        cols = set()
        rows = set()

        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == 0:
                    cols.add(j)
                    rows.add(i)

        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if i in rows or j in cols:
                    matrix[i][j] = 0
                    
        return matrix